### PR TITLE
Add tests for intersect/union, limit, group by, order by, variable, list exprs

### PR DESCRIPTION
**Pull request by jievince**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
By the way, fix https://github.com/vesoft-inc/nebula/issues/5112, fix https://github.com/vesoft-inc/nebula/issues/5114
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: tests/tck/features/match/PipeAndVariable.feature [hunk 1]
```diff
@@ -1,11 +1,142 @@
...
+      | a             |
+      | "Tony Parker" |
+      | "Tony Parker" |
+    When executing query:
+      """
+      $v1 = YIELD "Tony Parker" AS a;
+      $v2 = YIELD $v1.a AS a;
+      YIELD $v1.a, $v2.a
+      """
+    Then a SemanticError should be raised at runtime: Only one variable allowed to use.
```
Comment (Shylock-Hg) [inline tests/tck/features/match/PipeAndVariable.feature:112]: `Redefined variable: <var>`

**Review by nevermore3 — DISMISSED** [review]

File: tests/tck/features/match/PipeAndVariable.feature (hunk 1 above)
Comment (jievince) [reply tests/tck/features/match/PipeAndVariable.feature:112]: Redefined?

Comment (Shylock-Hg) [reply tests/tck/features/match/PipeAndVariable.feature:112]: Yes.

Comment (codesigner) [reply tests/tck/features/match/PipeAndVariable.feature:112]: add what varialbe is redefined in the error message

**Review by codesigner — COMMENTED** [review]
Good job

File: tests/tck/features/match/PipeAndVariable.feature (hunk 1 above)
Comment (jievince) [reply tests/tck/features/match/PipeAndVariable.feature:112]: I don't get it. This test case is not a redefined variable error, but multiple variable referenced in yield clause.

The case of redefined variable  is:
```
    When executing query:
      """
      $v1 = YIELD "Tony Parker" AS a;
      $v1 = YIELD "Tim Duncan" AS a;
      """
    Then a SemanticError should be raised at runtime: Variable `v1' already exists
```

Comment (Shylock-Hg) [reply tests/tck/features/match/PipeAndVariable.feature:112]: Ok, my mistake.

**Review by codesigner — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

File: tests/tck/features/set/Set.feature [hunk 2]
```diff
@@ -3,6 +3,98 @@
...
+      UNWIND [1,2] AS a RETURN a
+      UNION
+      UNWIND [2] AS a RETURN a
+      """
+    Then the result should be, in any order:
+      | a |
+      | 1 |
+      | 2 |
+
+  # cypher doesn't support intersect
```
Comment (Shylock-Hg) [inline tests/tck/features/set/Set.feature:79]: Uncomment it, and test the error.

Comment (jievince) [reply tests/tck/features/set/Set.feature:79]: This pr https://github.com/vesoft-inc/nebula/pull/5116  will support intersect/minus for cypher

**Pull request merged**